### ISSUE TITLE
Support /nix/store paths in manifest

### ISSUE THREAD

**Issue by mkenigs** [flox/flox#1764]
### Motivation

**When** Flox is missing software packaged correctly for my application (or just missing software)
**I need** to express my package / package group as a /nix/store path
**So I can** extend Flox with local Nix builds
**And** get working software in my environment locally

#### Manifest

Add a distinct descriptor schema for store paths alongside `ManifestPackageDescriptorCatalog` and `ManifestPackageDescriptorFlake`.

```
ManifestPackageDescriptorStorePath ::=  {
    store-path ::= STRING
    systems ::= [ List<SYSTEM_STRING> ]
}
```

#### Lockfile 

* Add a distinct descriptor schema for store paths alongside `LockedPackageCatalog` and `LockedPackageFlake`. 

```
ManifestPackageDescriptorStorePath ::=  {
    install_id ::= STRING
    store_path ::= STRING
    system ::= SYSTEM_STRING
}
```

* Locking is almost an identity function

#### CLI

* Extend `PackageToInstall` with StorePath variant.
* disambiguate path from either flake or catalog package arguments
    * `If let Ok(url) = Url::parse(arg) => parse flake` ( unchanged )
    * `If arg.starts_with(["./", "../", "~", "/"]) => parse store path
       * `if !store_path.canonicalize().starts_with("/nix/store")` => ParseError
    * `_ => catalog package` ( `{., .., ~. ε}` aren't allowed as catalog names
    * Note: referring to `./result` as `result` is _not_ supported for the lack of being able to disambiguate from catalog entries with the same name this early

#### Build

* check if store path is valid, err otherwise

**Comment by ghudgins** [comment]
how do we express this in the manifest given it's multi system

**Comment by limeytexan** [comment]
> how do we express this in the manifest given it's multi system

Just as you would for any package. Recall that a storepath could just be a data directory that someone added using the `nix store` command, so it's perfectly valid for a storepath to be useful on multiple systems.

That said, you're right that it's _less likely_ for a storepath installation to be valid on multiple systems, so a reasonable default would be to insert a limited `systems` attribute with an _imperative_ storepath installation.

In other words:
* if someone imperatively installs with `flox install /nix/store/x{32}-NAME_AND_VERSION_STRING` on `aarch64-darwin`
* then add something like this to the manifest:
    ```
    [install]
    NAME_AND_VERSION_STRING.pkg-path = "/nix/store/x{32}-NAME_AND_VERSION_STRING"
    NAME_AND_VERSION_STRING.systems = [ "aarch64-darwin" ]
    ```

**Comment by dcarley** [comment]
@ysndr is starting work on this and will summarise the [async refinement thread](https://flox-dev.slack.com/archives/C05P6A5J6U8/p1732025023666899) into the ticket description.
